src/main/java/org/pac4j/vertx/flow/AuthenticationFlow.java

```diff
@@ -0,0 +1,32 @@
+package org.pac4j.vertx.flow;
+
+import io.vertx.core.Vertx;
+import org.pac4j.core.client.Client;
+import org.pac4j.core.client.DirectClient;
+import org.pac4j.core.client.IndirectClient;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.vertx.VertxWebContext;
+
+import java.util.function.Consumer;
+
+/**
+ * Interface representing an authentication flow, effectively wrapping the differences required
+ * when initiating authentication for a direct or indirect client (the former actually needs to
+ * carry out authentication, the latter initiates the authentication flow by issuing a redirect
+ * response
+ * @author jez
+ */
```

Comment:
Just my personal feeling: I like to see `@since` annotation and complete name for the `@author` one.